docs/dev/cut-a-new-release.md

```diff
@@ -1,64 +1,230 @@
-# Glean Release Process
+# Glean development & release process
 
-These are the steps needed to cut a new release from latest master.
+Glean consists of multiple libraries for different platforms and targets.
+The main supported libraries are releases as one.
+The following document describes the development and release process.
+Development happens on the main repository <https://github.com/mozilla/glean>.
 
-1. Update the changelog.
+The development & release process roughly follows the [GitFlow model](https://nvie.com/posts/a-successful-git-branching-model/).
+
+> **Note:** The rest of this section assumes that `upstream` points to the `https://github.com/mozilla/glean` repository,
+> while `origin` points to the developer fork.
+> For some developer workflows, `upstream` can be the same as `origin`.
+
+**Table of Contents**:
+
+* [Development](#development)
+* [Regular release](#regular-release)
+* [Hotfix release for latest version](#hotfix-release-for-latest-version)
+* [Hotfix release for previous version](#hotfix-release-for-previous-version)
+
+## Published artifacts
+
+* The Kotlin libraries are published: [GitHub Releases](https://github.com/mozilla/glean/releases), [Mozilla Maven](https://maven.mozilla.org/?prefix=maven2/org/mozilla/telemetry/).
+* Python bindings are published on PyPI: [glean-sdk](https://pypi.org/project/glean-sdk/).
+* iOS framework artifacts: [GitHub Releases](https://github.com/mozilla/glean/releases).
+* Rust crates are published on crates.io: [glean-core](https://crates.io/crates/glean-core), [glean-ffi](https://crates.io/crates/glean-ffi).
+
+## Development
+
+All finished features for Glean land on the `master` branch.
+Features should be developed in feature branches and are sent as pull requests to the upstream repository.
+
+1. Clone the Glean repository:
+    ```
+    git clone https://github.com/your-name/glean && cd glean
+    ```
+2. Check out a new feature branch:
+    ```
+    git checkout -b your-feature-branch
+    ```
+3. Implement your feature and add information about it to `CHANGELOG.md` under the `Unreleased changes` headline.
+4. Push your branch:
+    ```
+    git push origin your-feature-branch
+    ```
+5. Create a new pull request: <https://github.com/mozilla/glean/compare/master...your-name:your-feature-branch?expand=1>
+
+The Glean maintainers will review your code and merge it when ready.
+Make sure all tests pass.
+
+## Regular Release
```

Comment:
"Regular" feels a bit vague.  Maybe "Making a feature release"?  That's not perfect either.  It's really a "non-hotfix release"...  Hmmm...   Naming things is hard.

Reply:
Maybe "Standard Release" works better?

Reply:
Yeah, "Standard Release" feels a little better than "Regular" which could imply "a regular cadence", which we don't do (or want to do).